ansible/library/ptf_portchannel.py

```diff
@@ -0,0 +1,193 @@
+    t = jinja2.Template(portchannel_conf_tmpl)
+    for conf in teamd_config:
+        with open(os.path.join(portchannel_conf_path, "{}.conf".format(conf["name"])), 'w') as fd:
+            fd.write(t.render(conf))
+
+
+def remove_teamd_conf(module, teamd_config):
+    for conf in teamd_config:
+        try:
+            os.remove(os.path.join(portchannel_conf_path, "{}.conf".format(conf["name"])))
+        except Exception:
+            pass
+
+
+def create_supervisor_conf(module, teamd_config):
+    t = jinja2.Template(portchannel_supervisord_conf_tmpl)
+    for conf in teamd_config:
+        with open(os.path.join(portchannel_supervisord_path, "portchannel-{}.conf".format(conf["name"])), 'w') as fd:
+            fd.write(t.render(conf))
+    refresh_supervisord(module)
+
+
+def remove_supervisor_conf(module, teamd_config):
+    for conf in teamd_config:
+        try:
+            os.remove(os.path.join(portchannel_supervisord_path, "portchannel-{}.conf".format(conf["name"])))
+        except Exception:
+            pass
+    refresh_supervisord(module)
+
+
+def enable_portchannel(module, teamd_config):
+    for conf in teamd_config:
+        for intf in conf["intfs"]:
+            exec_command(module, "ip link set dev {} down".format(intf))
+        exec_command(module, "supervisorctl start portchannel-{}".format(conf["name"]))
+        for count in range(0, 60):
+            time.sleep(1)
+            status = get_portchannel_status(module, conf["name"])
+            if u'RUNNING' == status:
+                break
+        assert u'RUNNING' == status
+        exec_command(module, "ip link set dev {} up".format(conf["name"]))
+
+
+
+def disable_portchannel(module, teamd_config):
+    for conf in teamd_config:
+        exec_command(module, cmd="supervisorctl stop portchannel-{}".format(conf["name"]), ignore_error=True)
+
+
+def setup_portchannel_conf():
+    try:
+        os.mkdir(portchannel_conf_path, 0755)
+    except OSError:
+        pass
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            cmd=dict(required=True, choices=['start', 'stop'], type='str'),
+            portchannel_config=dict(required=True, type='dict'),
+        ),
+        supports_check_mode=False)
+
+    cmd = module.params['cmd']
+    portchannel_config = module.params['portchannel_config']
+    teamd_config = parse_teamd_config(module, portchannel_config)
+
+    setup_portchannel_conf()
+
+    result = {}
+    try:
+        if cmd == 'start':
+            create_teamd_conf(module, teamd_config)
+            create_supervisor_conf(module, teamd_config)
+            enable_portchannel(module, teamd_config)
+        elif cmd == 'stop':
+            disable_portchannel(module, teamd_config)
+            remove_supervisor_conf(module, teamd_config)
+            remove_teamd_conf(module, teamd_config)
+    except Exception as e:
+        module.fail_json(msg=traceback.format_exc())
+
+    module.exit_json(**result)
```

Comment:
Is the `result` always an empty dict? I don't see any code updating it. Can you make it more meaningful?

Reply:
We don't have `get` function in this module, so I just remove this variable. Please check it.